src/main/webapp/app/shared/science/science.component.ts

```diff
@@ -0,0 +1,25 @@
+import { ScienceService } from 'app/shared/science/science.service';
+import { ScienceEventType } from 'app/shared/science/science.model';
+import { Directive } from '@angular/core';
+
+@Directive()
+export abstract class AbstractScienceComponent {
+    protected constructor(
+        protected scienceService: ScienceService,
+        private type: ScienceEventType,
+        private resourceId?: number,
+    ) {}
+
+    protected setResourceId(resourceId: number) {
+        this.resourceId = resourceId;
+    }
+
+    protected logEvent() {
+        if (this.scienceService.eventLoggingActive())
+            if (this.resourceId) {
+                this.scienceService.logEvent(this.type, this.resourceId).subscribe();
+            } else {
+                this.scienceService.logEvent(this.type).subscribe();
+            }
```

Comment:
This looks quite strange, I'm supprised the linter did not pick this up.
```suggestion
        if (this.scienceService.eventLoggingActive()) {
            if (this.resourceId) {
                this.scienceService.logEvent(this.type, this.resourceId).subscribe();
            } else {
                this.scienceService.logEvent(this.type).subscribe();
            }
       }
```